contrib/generic_proxy/filters/network/source/router/router.h

```diff
@@ -34,63 +35,80 @@ enum class StreamResetReason : uint32_t {
 };
 
 class RouterFilter;
+class UpstreamRequest;
 
-class UpstreamRequest : public Tcp::ConnectionPool::Callbacks,
-                        public Tcp::ConnectionPool::UpstreamCallbacks,
+/**
+ * We needn't to inherit from UpstreamManager interfaces because
+ * the Router::UpstreamManagerImpl is only used by Router::UpstreamRequest and
+ * only one request is allowed to be processed at the same time.
+ * We still inherit from UpstreamManagerImplBase to avoid code duplication.
+ */
+class UpstreamManagerImpl : public UpstreamManagerImplBase {
```

Comment:
As the comment here https://github.com/envoyproxy/envoy/pull/26618/files#r1163723493, if that works, then we needn't this UpstreamManagerImpl?